src/api/auth.go

```diff
@@ -101,7 +103,18 @@ func (a *Auth) checkPassword(username, password string) error {
 }
 
 func (a *Auth) deleteUser(username string) error {
-	result := a.db.Model(&User{}).Where(&User{Username: username}).Delete(&User{})
+	var adminUserCount int64
+	result := a.db.Model(&User{}).Where(&User{Role: "admin"}).Count(&adminUserCount)
+	if result.Error != nil {
+		log.Printf("Error retrieving admin user list from database: %s", result.Error)
+		return result.Error
+	}
+
+	if adminUserCount <= 1 {
```

Comment:
This will prevent to remove any user, as long as only one admin user is in the db.

Reply:
Nice find I fixed the query to allow deleting regular users if only one admin exists.